### PR TITLE
spline #368 New lineage overview query

### PR DESCRIPTION
**Pull request by wajda**

### CODE REVIEW

File: client-ui/src/app/effects/lineage-overview.effects.ts [hunk 1]
```diff
@@ -55,24 +54,29 @@ export class LineageOverviewEffects {
     )
 
     private getLineageOverview(executionEventId: string): Observable<LineageOverviewVM> {
-        return this.lineageOverviewControllerService.lineageUsingGET1Response(executionEventId).pipe(
+        return this.lineageOverviewControllerService
+          .lineageUsingGET1Response({executionEventId})
```
[the remark this answers is not in the document]
Comment (wajda) [reply client-ui/src/app/effects/lineage-overview.effects.ts]: This is a creation of JSON instance. The generated method signature looks this way: 
```typescript
lineageUsingGET1(params: LineageControllerService.LineageUsingGET1Params): __Observable<LineageOverview>
```
where `LineageUsingGET1Params` is:
```typescript
/**
   * Parameters for lineageUsingGET1
   */
  export interface LineageUsingGET1Params {

    /**
     * Max depth of the graph. ([Source] -> [App] -> [Target]) is considered one level
     */
    maxDepth?: number;

    /**
     * Execution event ID
     */
    executionEventId?: string;
  }
```

File: persistence/src/main/resources/AQLFunctions/observed_writes_by_read.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        FOR rds IN 2 OUTBOUND ${readEvent} progressOf, depends
+            LET maybeObservedOverwrite = SLICE(
+                (FOR wo IN 1 INBOUND rds writesTo
+                    FILTER !wo.properties.append
+                    FOR e IN 2 INBOUND wo executes, progressOf
+                        FILTER e.timestamp < readTime
+                        SORT e.timestamp DESC
+                        LIMIT 1
+                        RETURN e
+                ), 0, 1)
```
[the remark this answers is not in the document]
Comment (wajda) [reply persistence/src/main/resources/AQLFunctions/observed_writes_by_read.js]: The intention was to not having to deal with _null_'s in subsequent append/prepend, when there is no _overwrite_ found.
To maintain the chronological order of the returning result I am putting the found _write_ in the head of event sequence. However `UNSHIFT([1, 2], null)` gives me `[null, 1, 2]` while `APPEND([], [1, 2])` gives `[1, 2]` as expected. That's why I used `SLICE` and carry optional _overwrite_ as an array of 0 or 1 element.

File: client-ui/src/app/effects/lineage-overview.effects.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (leborgneromain) [reply client-ui/src/app/effects/lineage-overview.effects.ts]: ok ok, mb

File: persistence/src/main/resources/AQLFunctions/observed_writes_by_read.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (leborgneromain) [reply persistence/src/main/resources/AQLFunctions/observed_writes_by_read.js]: ok ok make sense

**Comment by wajda** [comment]
I forgot to add indices. Will make another push in a minute

**Pull request merged**